lib/exports/mixins.less

```diff
@@ -110,6 +110,40 @@
 }
 
 
+/**
+ * Focus styles for the given context.
+ *
+ * Usage example:
+ * .focus-styles(true, true);
+ *
+ * @inset: boolean - whether the focus style be placed inside the element.
+ * @border: boolean - whether the element's border color change to match the focus style.
+ */
+.focus-styles(@inset: false, @border: false) {
```

Comment:
> When I am in a component less file and I see `.focus-style(true, true)` I have no clue what true and true means and I would need to come here to figure this out. Something like this would be more readable in my opinion:
> 
> ```css-less
> .component {
>     .focus-styles({
>         inset: true;
>         border: false;
>     });
> }
> ```
> 
> What do you think?
> 
> I believe it could be achieved with something like this but I have not tested it myself. 🙂
> 
> ```css-less
> .focus-styles(@options: {}) {
>     @inset: extract(@options, inset);
>     @border: extract(@options, border);
>     
>     ...
> ```


I would like that, but Less does not support object literal arguments 🙁 I remember wanting this when I was creating all the color mixins and had to include one property per argument.

It tried your suggestion (and a few variations) in the Less Preview online playground:

```Less
.focus-styles(@options: {}) {
    @inset: extract(@options, inset);
    @border: extract(@options, border);

    & when (@inset = false) {
        color: red;
    }
    & when (@inset = true) {
        color: blue;
    }
    & when (@border = false) {
        border-color: yellow;
    }
    & when (@border = true) {
        border-color: green;
    }
}

.component {
    background: purple;

    .focus-styles({
        inset: false;
        border: false;
    });
}
```

and got the following output:

```css
.component {
  background: purple;
}
```

FWIW, I originally included arguments that accepted strings like `bordered/unbordered` and `inside/outside`, but I scrapped them in favor of booleans. Would you prefer something like that? I'm happy to explore it if you think that's better.

Reply:
I think I am ok to stick to booleans. I guess Less is starting to show its age. 
For my curiosity if you would start today how would you recommend to accomplish this behavior? PlainCSS + PostCSS plugin?

Reply:
That's a good question! My gut says we could do pretty well with PostCSS + a few plugins (probably for [nesting](https://www.npmjs.com/package/postcss-nesting), [Sass/Less-like variables](https://github.com/postcss/postcss-simple-vars), and [mixins](https://github.com/postcss/postcss-mixins)) but I haven't researched this much so I'm not 100% sure it's the way to go.

> PlainCSS + PostCSS

I'm guessing by PlainCSS, you intended plain as in _vanilla_ and not [PlainCSS](https://ilyadzh.github.io/plain-css), right?

Reply:
Yes. vanilla.  Of course there is a PlainCSS framework. 😅